docs/shape_batch_size_and_layout.md

```diff
@@ -31,17 +31,33 @@ In case the model can't be reshaped, it will remain in the original parameters a
+In case this is not possible, or you already have model with NCHW layout and are not willing to re-export it, OpenVINO™ Model Server allows changing input/output layout at runtime with `--layout` parameter via CLI or `config.json`. Please note that it modifies the model by adding transposition operations as pre-processing step.
 
-Layout parameter is optional. By default layout is inherited from OpenVINO™ model. You can specify layout during conversion to IR format via Model Optimizer. You can also use this parameter for ONNX models.<br>
+Layout parameter is optional. By default layout is inherited from OpenVINO™ model. You can also use this parameter for ONNX models. In case no layout was specified during model export, default layout is `N...` which means that there is only one known dimension which is batch (`N`) and it is on first position.
 
-Layout change is only supported to `NCHW` or `NHWC`. You can specify 2 forms of values:
-  * string - either `NCHW` or `NHWC`; applicable only for models with single input tensor
-  * dictionary of strings - e.g. `{"input1":"NHWC", "input2":"NCHW", "output1":"NHWC"}`; allows to specify layout for multiple inputs and outputs by name.
+Layout change is supported for variety of combinations with `N`, `C`, `H`, `W`, `D`, `?` and letter `...` combinations. Each dimension can appear only once. The exception is `?` which can appear multiple times meaning that there is unknown dimension on given position.
 
-After the model layout is changed, the requests must match the new updated shape in order NHWC instead of NCHW. For NCHW inputs it should be: `(batch, channels, height, width)` but for NHWC this is: `(batch, height, width, channels)`.
+Examples:
+- `NHW` - means there are 3 dimensions: _batch_, _height_ and _width_.
+- `N??C` - means there are 4 dimensions: first being _batch_, 4th being the _channels_ and 2 unknown dimensions on second and third position
+- `NC...W` - means there are undefined amount of dimensions: first being _batch_, second being the _channels_ and the last dimension is _width_.
 
-Changing layout is not supported for models with input names the same as output names.<br>
+You can specify 2 forms of values:
+  * string - e.g. `NCHW` (which expands to `NCHW:NCHW`) or `NHWC:NCHW`; applicable only for models with single input tensor
+  * dictionary of strings - e.g. `{"input1":"NHWC:NCHW", "input2":"NHWC:NCHW", "output1":"CN:NC"}`; allows to specify layout for multiple inputs and outputs by name.
+
+The `LAYOUT_A:LAYOUT_B` notation means `[A=input data layout : B=expected layout by model]`. If both layouts are different, OpenVINO™ takes care of applying additional transposition step.
```

Comment:
in the model optimizer documentation it is described as target layout and source layout. It might be better from layout_a and layout_b

Reply:
In the same sentence I specify that A=input data layout and B=expected layout by model which describe the setting even more than target/source words